### DISCUSSION TITLE
How to make localleader keymaps file-type specific

### DISCUSSION THREAD

**Discussion by weberam2** [LazyVim/LazyVim#4637] [Q&A]
I feel like I have tried so many different routes to get localleader keys to be file-type specific

Does anyone have a working example for a plugin that loads for a specific filetype? I thought looking at the https://www.lazyvim.org/extras/lang/r would help, but trying to copy that example didn't work

Cheers

**Comment by jyuan0** [comment]
> Does anyone have a working example for a plugin that loads for a specific filetype?

The rustaceanvim plugin in the Rust extra should lazy-load on Rust files: https://github.com/LazyVim/LazyVim/blob/2f6c1f60834108359b3404748453a65843909a03/lua/lazyvim/plugins/extras/lang/rust.lua#L47

----

> I feel like I have tried so many different routes to get localleader keys to be file-type specific

Have you tried setting `ft` in the mapping in `keys`? E.g.: Some of the octo.nvim plugin keymaps are file-type specific: https://github.com/LazyVim/LazyVim/blob/2f6c1f60834108359b3404748453a65843909a03/lua/lazyvim/plugins/extras/util/octo.lua#L25

**Reply by weberam2** [reply]
This is a great place to start. I will try this when I get home

**Reply by weberam2** [reply]
I have tried some test cases the Octo way and that seems to be getting me somewhere. Thanks again!
